### ISSUE TITLE
Talisman fails silently if parts of the .talismanrc are not valid yaml

### ISSUE THREAD

**Issue by karlbrown-va** [thoughtworks/talisman#454]
**Describe the bug**
`.talismanrc` is supposed to be a yaml file. However, the filename does not end in .yaml, certain editors may not detect it and use yaml settings, and additionally, yaml linters will not detect it by default to find errors.

This can lead to silent failures - for example if a `tab` is added to one of the lines. Further lines in the file (such as custom_patterns) could be ignored, leading to password leakage.

**To Reproduce**
1. Create a new git repository and add a test file
2. Add a custom pattern to .talismanrc: 
```
custom_patterns:
  - 'pwd_[a-z]{8,20}'
```
4. Add a string matching this pattern to the file
5. `git add .` and then `talisman -l debug --githook pre-commit` and confirm the secret is detected and the custom pattern was used
6. Replace the spaces in .talismanrc with tabs
7. `git add .` and then `talisman -l debug --githook pre-commit` and confirm the secret no longer detected

**Expected behavior**
* Talisman should detect a malformed `.talismanrc.yaml` file, or perform some basic sanity checks on the file
* `.talismanrc` should be renamed as `.talismanrc.yaml` so it is detected by yamllinters and editors.
* The samples of `.talismanrc.yaml` should be updated to be proper yaml (with correct indenting)
* The output of the checksum tool should be properly indented (currently it is not indented - e.g. 
```
fileignoreconfig:
- filename: readme.md
  checksum: xxx
version: ""
```
should be
```
fileignoreconfig:
  - filename: readme.md
    checksum: xxx
version: ""
```

**Desktop (please complete the following information):**
 - OS: OSX

**Comment by tw-owen-nelson** [comment]
After #458 and #461 talisman will fail on a malformed `.talismanrc` file (i.e. if it cannot unmarshal the content into our expected struct). 

I'm open to checking for a `.talismanrc.yaml` file in addition to `.talismanrc`.

Indenting and prefixing have less clear value to me. Is this so that the generated `.talismanrc[.yaml]` complies with yaml linters? We currently leave this formatting to go-yaml/yaml's marshaller and would need to reformat ourselves to achieve that.

**Comment by karlbrown-va** [comment]
Yes, I think if it's yaml it should comply with the linter. Would be great if you accepted .yaml / .yml files (both prefixes seem to be allowed in the yaml world)
